api/controllers/v0.3/ILSHelper.js

```diff
@@ -0,0 +1,390 @@
+      // home_lib_as_field,
+      patron.policy.agency,
+    ];
+
+    if (patron.worksInCity) {
+      fields.push(fieldData["workAddress"]);
+      fields.push(note_as_field);
+    }
+
+    if (this.longExpiration(patron)) {
+      fields.push(this.longExpirationAsField(patron));
+    } else {
+      patron.set_temporary();
+      fields.push(this.shortExpirationAsField(patron));
+    }
+
+    // Add required/optional fields according to policy.
+    if (patron.barcode) {
+      fields.push(fieldData["barcode"]);
+    }
+    if (patron.email) {
+      fields.push(fieldData["email"]);
+    }
+    if (patron.birthdate) {
+      fields.push(fieldData["birthdate"]);
+    }
+
+    return {
+      patronFields: {
+        patron_field: fields,
+      },
+      patronID: patron.patronId,
+    };
+  }
+
+  blankField() {
+    return { fieldTag: "", marcTag: 0, value: "", value_is_binary: false };
+  }
+
+  customField(fieldTag, value, marcTag = 0) {
+    let field = this.blankField();
+    field["field_tag"] = fieldTag;
+    field["value"] = value;
+    field["marc_tag"] = marcTag;
+    return field;
+  }
+
+  expirationAsField(time) {
+    // TODO convert date object
+    let expiration = time.strftime("%Y%m%d000000");
+    return this.customField(
+      IlsHelper.EXPIRATION_FIELDTAG,
+      expiration,
+      IlsHelper.STRING_MARCTAG
+    );
+  }
+
+  longExpirationAsField(patron) {
+    let timeNow = Date.now();
+    return this.expirationAsField(
+      `${timeNow}${parseInt(patron.policy.card_type["standard"], 10)}`
+    );
+  }
+
+  shortExpirationAsField(patron) {
+    let timeNow = Date.now();
+    return this.expirationAsField(
+      `${timeNow}${parseInt(patron.policy.card_type["temporary"], 10)}`
+    );
+  }
+
+  // Internal: Create dynamic field-creation method.
+  //
+  // field - A String representing the base field name. <field>_FIELDTAG
+  //         must be set in IlsHelper. Can have any capitalization.
+  //
+  // value - A primitive type Object. May be nil. If nil, DEFAULT_<field>
+  //         constant must be set in IlsHelper
+  //
+  // If the field requires the STRING_MARCTAG, it must be included in the
+  // IlsHelper.WITH_MARK_TAG array.
+  //
+  // Signature
+  //
+  //   <field>_as_field([value])
+  //
+  // field - a field name
+  //
+  // Returns a Hash representing the custom field.
+  dynamicFieldAsField(field, value) {
+    try {
+      field = field.toUpperCase();
+      // get these values from the class:
+      const fieldTag = `${field}_FIELDTAG`;
+      value = value || `DEFAULT_${field}`;
+      let marcTag = IlsHelper.WITH_MARCTAG.includes(field.downcase)
+        ? IlsHelper.STRING_MARCTAG
+        : null;
+      return this.customField(fieldTag, value, marcTag);
+    } catch (e) {
+      // throw new NotImplementedError(`//${field} field details have not been set`);
+      throw new Error(`//${field} field details have not been set`);
+    }
+  }
+}
+
+IlsHelper.MINOR_AGE = 11;
```

Comment:
It might not be worth changing this, if it's used in a bunch of different places...but maybe call this `CHILD_AGE` or something.  It's confusing having "minor" mean "under 12."

Reply:
That's true but it could also be convention used by the ILS so I'll double-check with the ILS team first. It might make it easier to make calls if we use the convention the ILS expects rather than having to convert and remember both those variable names.